### ISSUE TITLE
`reportUnusedFunction`: false-positive when the function is only used in doctests

### ISSUE THREAD

**Issue by AlexWaygood** [microsoft/pyright#4692]
**Describe the bug**

flake8-pyi has [multiple doctests](https://github.com/PyCQA/flake8-pyi/blob/7dd439880361ee467cbd237a6af2eb3528c6317d/pyi.py#L327-L335) that are an important parts of the test suite, acting as unit tests for discrete parts of the code. Most of these doctests use the helper function [`_ast_node_for`](https://github.com/PyCQA/flake8-pyi/blob/7dd439880361ee467cbd237a6af2eb3528c6317d/pyi.py#L298-L302). However, since this helper function is only used in doctests, pyright falsely concludes that this private function is unused.

**To Reproduce**

Run pyright on this file:

```python
# pyright: reportUnusedFunction=true

def _foo(): pass


class Bar:
    """
    >>> _foo()
    """
```

**Expected behavior**

No error should be reported: the function `_foo` is used in a doctest.

**Actual behaviour**

The `reportUnusedFunction` diagnostic reports that the function `_foo` is unused.

**Comment by erictraut** [comment]
The current behavior is correct. Pyright doesn't parse text in comments, so a reference to a symbol within a docstring is not considered a reference. You can disable the `reportUnusedFunction` check locally to the file if you would like.
